src/masternodeman.cpp

```diff
@@ -659,6 +658,9 @@ int CMasternodeMan::ProcessMNBroadcast(CNode* pfrom, CMasternodeBroadcast& mnb)
         return 33;
     }
 
+    // now that did the basic mnb checks, can add it.
+    mapSeenMasternodeBroadcast.emplace(mnbHash, mnb);
```

Comment:
wouldn't it be better to update the map inside `if (mnb.CheckInputsAndAdd(nDoS)) {}` few lines below?

Reply:
yeah, I actually moved it there at the beginning but reverted it because there were other checks depending on it.. need to double check the flow again. 
We should definitely decouple properly and unify `CheckAndUpdate` and `CheckInputsAndAdd` in the future.